test/extended/client.go

```diff
@@ -0,0 +1,240 @@
+package extended
+
+//go:generate go get github.com/go-bindata/go-bindata/go-bindata
+//go:generate go-bindata -nometadata -pkg $GOPACKAGE -prefix data data/...
+//go:generate gofmt -s -l -w bindata.go
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"os"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/checks"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/client-go/util/retry"
+	kaggregator "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+)
+
+// Interface exposes the methods a client needs to implement
+// for the syncing process of the addons.
+type Interface interface {
```

Comment:
I don't think this is necessary or appropriate.  This is test code and we're supposing `client` will be a singleton.  just use a struct with public members for all the clients you need.  Start with https://github.com/kubernetes/client-go/blob/master/kubernetes/clientset.go#L127 because it contains nearly everything.